corehq/apps/hqwebapp/tasks.py

```diff
@@ -93,6 +95,7 @@ def send_mail_async(self, subject, message, from_email, recipient_list,
             from_email=from_email,
             to=filtered_recipient_list,
             headers=headers,
```

Comment:
Can you investigate these headers?  How many of them are gateway specific?  I wonder whether `connection` encapsulates enough of this infrastructure.  I think `from_email` and `headers` will likely need to be part of what's provided by the custom backend.

Reply:
I looked into it, there's 3 other headers:
- `X-COMMCAREHQ-MESSAGE-ID` This is the id of the `MessagingSubEvent` item so wouldn't be specific to the custom backend
- `Return-Path` This is the return email used to forward BOUNCE & COMPLAINT notifications. This would be specific to the custom backend if partners wanted to monitor their bounces/complaints and want a custom email for this. For now, I don't think that's needed
- `X-SES-CONFIGURATION-SET` - This would be specific to the custom backend. So that SNS can probably reach our log_email_event endpoint. Still in the process of figuring out how this works but the SNS endpoint secret field that Akash added should be all that's needed.

Reply:
Thanks, that's helpful.  I wasn't familiar with `Return-Path`, but [I found this article on it](https://sendgrid.com/blog/what-is-return-path/).  Sounds like it's important in establishing credibility of the email sender.  I agree that we can leave `X-COMMCAREHQ-MESSAGE-ID` in for everything.  It should be fine to get email working end-to-end without the other stuff for a first pass, though I think we will need to get a configurable `Return-Path` set up eventually, as well as the SNS stuff.